src/providers/realtime-telemetry-provider.js

```diff
@@ -23,12 +23,107 @@
 import {
     idToQualifiedName,
     qualifiedNameToId,
-    getValue
+    getValue,
+    addLimitInformation
 } from '../utils.js';
 
 const WS_IDLE_INTERVAL_MS = 10000;
 const FALLBACK_AND_WAIT_MS = [1000, 5000, 5000, 10000, 10000, 30000];
 
+/* CSS classes for Yamcs parameter monitoring result values. */
+const MONITORING_RESULT_CSS = {
+    'WATCH': 'is-limit--yellow',
+    'WARNING': 'is-limit--yellow',
+    'DISTRESS': 'is-limit--red',
+    'CRITICAL': 'is-limit--red',
+    'SEVERE': 'is-limit--red'
+}
+
+/* CSS classes for Yamcs range condition values. */
+const RANGE_CONDITION_CSS = {
+    'LOW': 'is-limit--lwr',
+    'HIGH': 'is-limit--upr'
+}
+
+
+/*
+ * Implements an OpenMCT limit evaluator that uses limit information passed
+ * from either the historical or realtime telemetry source to create limit
+ * information for OpenMCT.
+ */
+class LimitEvaluator {
+
+    /*
+     * Evaluates a telemetry point for limit violations.
+     *
+     * Parameters:
+     *     datum: the telemetry point data from the historical or realtime
+     *            plugin
+     *     valueMetadata: metadata about the telemetry point, not used
+     *
+     * Returns:
+     *     an object with CSS class information, a violation name, and
+     *     range information, if there is a limit violates, or nothing,
+     *     if the value is within limits.
+     *
+     * The datum parameter may include the following information:
+     *     monitoringResult: the Yamcs limit monitoring result, if any
+     *     rangeCondition: the Yamcs range condition (LOW/HIGH), if any
+     *     alarmRange: an array of alarm ranges for different monitoring
+     *         results, or omitted, if no alarm ranges are defined
+     */
+    evaluate(datum, valueMetadata) {
+        if (datum.monitoringResult
+            && datum.monitoringResult in MONITORING_RESULT_CSS) {
+            let obj = {
+                cssClass: MONITORING_RESULT_CSS[datum.monitoringResult],
+                name: datum.monitoringResult
+            }
+            if (datum.rangeCondition
+                && datum.rangeCondition in RANGE_CONDITION_CSS) {
+                obj.cssClass += ' '
+                obj.cssClass += RANGE_CONDITION_CSS[datum.rangeCondition]
+                this.addLimitRange(datum, datum.monitoringResult, obj)
+            }
+            return obj
+        }
+    }
+
+    /*
+     * Adds limit range information to an object based on the monitoring
+     * result.
+     */
+    addLimitRange(datum, result, obj) {
```

Comment:
Maybe we could rename `obj` to `evaluationResult` or something? I was a little confused initially about what this was, and thought at first it was an Open MCT domain object.

Reply:
Will fix.